src/Uno.UI/UI/Xaml/Controls/Repeater/RecyclePool.cs

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Windows.UI.Xaml;
+
+namespace Microsoft.UI.Xaml.Controls
+{
+	public partial class RecyclePool
+	{
+		private readonly Dictionary<string, List<ElementInfo>> m_elements = new Dictionary<string, List<ElementInfo>>();
+
+		private struct ElementInfo
+		{
+			public ElementInfo(UIElement element, IPanel owner)
+			{
+				Element = element;
+				Owner = owner;
+			}
+
+			public UIElement Element { get; }
+			public IPanel Owner { get; }
+		};
+
+		public void PutElement(UIElement element, string key)
+			=> PutElementCore(element, key, null /* owner */);
+
+		public void PutElement(UIElement element, string key, UIElement owner)
+			=> PutElementCore(element, key, owner);
+
+		public UIElement TryGetElement(string key)
+			=> TryGetElementCore(key, null /* owner */);
+
+		public UIElement TryGetElement(string key, UIElement owner)
+			=> TryGetElementCore(key, owner);
+
+		protected virtual void PutElementCore(UIElement element, string key, UIElement owner)
+		{
+			var winrtKey = key;
+			var winrtOwner = owner;
+			var winrtOwnerAsPanel = EnsureOwnerIsPanelOrNull(winrtOwner);
+
+			ElementInfo elementInfo = new ElementInfo(element, winrtOwnerAsPanel);
+
+			if (m_elements.TryGetValue(winrtKey, out var infos))
+			{
+				infos.Add(elementInfo);
+			}
+			else
+			{
+				var pool = new List<ElementInfo>();
+				pool.Add(elementInfo);
+				m_elements[winrtKey] = pool;
+			}
+		}
+
+		protected  virtual UIElement TryGetElementCore(string key,  UIElement owner)
+		{
+			if (m_elements.TryGetValue(key, out var elements))
+			{
+				if (elements.Count > 0)
+				{
+					ElementInfo elementInfo = default;
```

Comment:
Codacy found an issue: [Remove this useless assignment to local variable 'elementInfo'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5863680)